Src/FluentAssertions/Primitives/StringAssertions.cs

```diff
@@ -507,8 +507,9 @@ public AndConstraint<TAssertions> NotMatch(string wildcardPattern, string becaus
                     .ForConstraint(occurrenceConstraint, actual)
                     .UsingLineBreaks
                     .BecauseOf(because, becauseArgs)
-                    .FailWith($"Expected {{context:string}} to match regex {{0}} {{expectedOccurrence}}{{reason}}, but found it {actual.Times()}.",
-                        regexStr);
+                    .FailWith($"Expected {{context:string}} to match regex {{0}} {{expectedOccurrence}}{{reason}}, " +
+                    $"but found it {actual.Times()} in {{1}}.",
+                        regexStr, Subject);
```

Comment:
Should we use the same pattern for including `Subject` as `Contain(string expected, OccurrenceConstraint occurrenceConstraint` and `ContainEquivalentOf(string expected, OccurrenceConstraint occurrenceConstraint` does?

Reply:
Hmm.. fair point 🤔

But what if you have a really long subject? I think this will mess up the failure message more than just appending the subject.

Reply:
Another posibility is to insert a line break before and/or after subjects contents.

Reply:
Maybe we should consider to update `StringValueFormatter` to add two newlines when the string exceeds a certain length?

Reply:
Should this be part of this PR?

Reply:
Maybe not. Probably not.

Reply:
I recall our failure messages in general could use an overhaul to break long lines.
But I agree, that should not be part of this PR.
Let's align with the other methods for now in this PR.